src/WebForms/UI/Control.cs

```diff
@@ -927,7 +929,15 @@ internal RouteCollection RouteCollection
                 _occasionalFields.RareFields == null ||
                 _occasionalFields.RareFields.RouteCollection == null)
             {
-                return RouteTable.Routes;
+                
+               if (Context !=null)
+                {
+                    return Context.AsAspNetCore().RequestServices.GetRequiredService<IOptions<HttpHandlerOptions>>().Value.Routes;
```

Comment:
should we cache the value locally ? (instead of calling into DI again/again)

Reply:
yeah, as long as the context will remain unique for the lifetime of the control